File: fixtures/dart/post_with_data_raw.dart
import 'package:http/http.dart' as http;

void main() async {
  var data = 'msg1=wow&msg2=such&msg3=@rawmsg';

  var url = Uri.parse('http://example.com/post');
  var res = await http.post(url, body: data);
  if (res.statusCode != 200) throw Exception('http.post error: statusCode= ${res.statusCode}');
  print(res.body);
}

File: fixtures/matlab/post_with_data_raw.m
%% Web Access using Data Import and Export API
uri = 'http://example.com/post';
body = 'msg1=wow&msg2=such&msg3=@rawmsg';
response = webwrite(uri, body);

%% HTTP Interface
import matlab.net.*
import matlab.net.http.*
import matlab.net.http.io.*

uri = URI('http://example.com/post');
body = FormProvider('msg1', 'wow', 'msg2', 'such', 'msg3', '@rawmsg');
response = RequestMessage('post', [], body).send(uri.EncodedURI);
